src/main/kotlin/com/esops/repository/OrderRepository.kt

```diff
@@ -0,0 +1,67 @@
+package com.esops.repository
+
+import com.esops.entity.BuyOrderComparator
+import com.esops.entity.Order
+import com.esops.entity.OrderType
+import com.esops.entity.SellOrderComparator
+import jakarta.inject.Singleton
+import java.math.BigInteger
+import java.util.*
+
+@Singleton
+class OrderRepository {
+
+    private var orders = HashMap<String, HashMap<Long, Order>>()
+    private var buyOrderQueue = PriorityQueue(BuyOrderComparator)
+    private var sellOrderQueue = PriorityQueue(SellOrderComparator)
+    private var orderIDCounter: Long = 0
+
+    fun incrementOrderIDCounter() {
+        orderIDCounter++
+    }
+
+    fun getOrderIDCounter(): Long {
+        return orderIDCounter
+    }
+
+    fun initializeOrderMapIfEmpty(username: String) {
+        if (orders[username].isNullOrEmpty())
+            orders[username] = HashMap()
+    }
+
+    fun getOrderByUsername(username: String): List<Order> {
+        return orders[username]!!.values.toList()
+    }
+
+    fun addOrder(username: String, order: Order) {
+        orders[username]?.set(orderIDCounter, order)
+    }
+
+    fun getOrderQueue(type: OrderType): PriorityQueue<Order> {
+        if(type == OrderType.BUY)   return buyOrderQueue
```

Comment:
this function should be an expression

Reply:
Created separate functions for buy order queue and sell order queue